cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -101,7 +101,6 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
   define_option_string(ARROW_SIMD_LEVEL
                        "SIMD compiler optimization level"
                        "SSE4_2" # default to SSE4.2
-                       "NONE"
```

Comment:
I personally prefer to keep the NONE(zero) level here though it may duplicate to ARROW_USE_SIMD. Level usually start from zero.

Reply:
Thanks for review. NONE restored.